gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/tasks/CreateManifestTask.groovy

```diff
@@ -66,6 +85,104 @@ class CreateManifestTask extends DefaultTask {
 
     @TaskAction
     void createManifest() {
+        def jsonMapper = new ObjectMapper()
+                .setSerializationInclusion(JsonInclude.Include.NON_NULL)
+                .setPropertyNamingStrategy(new KebabCaseStrategy())
+
+        Set<RecommendedProductDependency> allRecommendedProductDeps = []
+        Map<String, Set<RecommendedProductDependency>> allRecommendedDepsByCoord = [:]
+        Map<String, String> mavenCoordsByProductIds = [:]
+
+        productDependenciesConfig.resolvedConfiguration.resolvedArtifacts.each { artifact ->
+            def coord = identifierToCoord(artifact.moduleVersion.id)
+
+            def manifest
+            try {
+                def zf = new ZipFile(artifact.file)
+                def manifestEntry = zf.getEntry("META-INF/MANIFEST.MF")
+                manifest = new Manifest(zf.getInputStream(manifestEntry))
+            } catch (IOException e) {
+                logger.warn("IOException encountered when processing artifact '{}', file '{}'", coord, artifact.file, e)
+                return
+            }
+
+            def pdeps = manifest.getMainAttributes().getValue(SLS_RECOMMENDED_PRODUCT_DEPS_KEY)
+
+            if (pdeps == null) {
+                logger.debug("No pdeps found in manifest for artifact '{}', file '{}'", coord, artifact.file)
+                return
+            }
+
+            def recommendedDeps = jsonMapper.readValue(pdeps, RecommendedProductDependencies.class)
+
+            if (!allRecommendedDepsByCoord.containsKey(coord)) {
+                allRecommendedDepsByCoord.put(coord, new HashSet<RecommendedProductDependency>())
```

Comment:
Then I'd need to pull in guava as a dependency :/ have had issue in the past when plugins have done this... your call